### ISSUE TITLE
Ugrading to 2.12.22 is catastrophic!

### ISSUE THREAD

**Issue by NeedleFactory** [texstudio-org/texstudio#916]
SUMMARY: On a MacBookPro: upgrading TeXstudio 2.12.16 to 2.12.22 broke something drastic; after my retreat to 2.12.16 it is still broken.

DETAILS: 
(1) Compiling fails, with this message:
Could not start Build & View:PdfLaTeX: a warning DLOG appears with three lines:
     Could not start Default Compiler:PdfLaTeX:
     pdflatex -synctex=1 -
     interaction=nonstopmode "Axioms v68f".tex
- TeXstudio:  2.12.22
- Qt:  5.12.3, compiled with Qt 5.12.3 R
- OS: MacOS 19.15.2 (Catalina) 
- TeX distribution: ?? (it was downloaded by TeXstudio 2.12.16)

### Expected behavior
  what I'm used to
### Actual behavior
   document that compiled two days ago now fails;
   even after rebooting the computer!

**Comment by NeedleFactory** [comment]
I just found the "texput.log", it says:

This is pdfTeX, Version 3.14159265-2.6-1.40.20 (TeX Live 2019) (preloaded format=pdflatex 2019.5.8)  31 JAN 2020 15:54
entering extended mode
 restricted \write18 enabled.
 %&-line parsing enabled.
**"Axioms v68f.tex"

! Emergency stop.
<*> "Axioms v68f.tex"
                     
*** (job aborted, file error in nonstop mode)

 
Here is how much of TeX's memory you used:
 3 strings out of 492616
 108 string characters out of 6129482
 57117 words of memory out of 5000000
 4025 multiletter control sequences out of 15000+600000
 3640 words of font info for 14 fonts, out of 8000000 for 9000
 1141 hyphenation exceptions out of 8191
 0i,0n,0p,1b,6s stack positions out of 5000i,500n,10000p,200000b,80000s
!  ==> Fatal error occurred, no output PDF file produced!

**Comment by NeedleFactory** [comment]
I have computer backups of when TeXstudio (v2.12.16) was working fine.
Other than the application itself, what files (in what folders) ought I restore?

**Comment by MeanSquaredError** [comment]
You can go to Help/Check LaTeX Installation
Then once it generates the system report search for the setting called `Setting file` which should look like:
`Setting file: /home/john/.config/texstudio/texstudio.ini`
(here `/home/john/.config/` is the path to the system-specific settings directory, so it will be different in your case)

The part which says `/home/john/.config/texstudio/` is the directory where all kinds of settings and runtime-generated data is written. You can try renaming that directory to something else, e.g. to
`/home/john/.config/texstudio.old`
just to prevent TXS from finding it and then see if reinstalling TXS 2.12.16 again fixes the problems.

**Comment by sunderme** [comment]
are you sure that not the tex file or the texlive installation is not working ?
You should try to compile the file from a terminal first. If that works, you can look into what may be wrong with txs.

**Comment by NeedleFactory** [comment]
MeanSquaredError: thanks for helping.
I still have 2.12.16 running on an older Mac.
Per you suggestion, I ran 'Help/Check LaTeX Installation' on both computers.

(A) Results on each are roughly similar EXCEPT the older computer's first data line begins:
   Apple_PubSub_Socket_Render=/private/tmp/com.apple....
and the report on the newer computer contains no such line.
I find this suspicious as it appears that no "rendering" is occurring. 

(B) On neither report is there a line containing the word 'config'
Otherwise the reports have the same format, with lines beginning with these words:
DISPLAY, HOME, LOGNAME, PATH, SHELL, SSH_AUTH_SOCK, TMPDIR, USER,
XPC_FLAGS, SPC_SERVICE_NAME,,++CF_user_TEXT_ENCODING.

sunderme: thanks to you also.
I have many old versions of the source LaTeX file; and a version that works on
the old installation will not work on the new; thus I suspect
the new installation may be corrupted. I don't know how to re-install;
for the upgrade I merely replaced the old app with the new one (in the Applications folder).
I have forgotten the details of my original installation last August; IIRC I merely began
running the app and it installed whatever it needed itself. I could be wrong.

**Comment by MeanSquaredError** [comment]
@NeedleFactory 
The word `config` is not important (in my case `.config` is just the name of the directory where Fedora Linux stores the user settings).
What is important is the line starting with `Setting file`. Is there such a line in the system report that your installation of TeXstudio has created after you selected `Help/Check LaTeX Installation` ?

Indeed, as sunderme suggested, it looks like a problem with the installation of LaTeX. Renaming (or deleting altogether) the settings for TeXstudio is one way to check that. So I would suggest the following steps:

1. Find out the location of the settings file (as I described above).
2. Uninstall TeXstudio.
3. Rename (renaming is better than deleting) the TeXstudio settings directory (not the user's settings directory, just the texstudio settings directory).
4. Install TeXstudio 2.12.16.

If the problem persists after installing TeXstudio 2.12.16 then most likely it is like sunderme said a problem with a damaged LaTeX installation. You can verify that out as he suggested by opening a command prompt/shell window and running 
```
cd /directory/where/is/your/tefile
pdflatex 'Axioms v68f.tex'
```
If that command terminates incorrectly then it would be another proof of a problem with the LaTeX installation.

**Comment by NeedleFactory** [comment]
@MeanSquaredError, @sunderme
I found _no_ line containing `setting` in either the old or the new system report.
Eventually, I found out why: If the upper left portion of the screen contains the input file,
then (in the Mac version) then `Help/Check LaTeX Installation` erases that portion,
and a brief summary appears below (with no mention about a setting file).
Additional checks give the same result, although (now I notice) the full report
flashes briefly in the upper left portion before being erased.

I have now succeeded in getting 2.12.22 working as it should.
(It may have been a cockpit error too difficult to explain.)
I am very grateful for your help, both of you; I have benefited from your advice.
Although _my_ problem is now resolved, I'll leave it to others to "close" the issue, because
the Macintosh implementation may have a bug in `Help/Check LaTeX Installation`,
as I described above.

Again, thanks to both!
